### PR TITLE
Add DroneDB integration (download & share)

### PR DESCRIPTION
**Pull request by HeDo88TH**
DroneDB settings page:

login

Dataset choice and upload

Schermata del 2022-01-17 02-08-59

Now working on refining the process and on DroneDB upload result files.

### CODE REVIEW

File: coreplugins/dronedb/api_views.py [hunk 1]
```diff
@@ -273,7 +273,17 @@ def ddb_cleanup(sender, task_id, **kwargs):
     # When a task is re-processed, make sure we can re-share it if we shared a task previously
 
     logger.info("Cleaning up DroneDB datastore for task {}".format(str(task_id)))
-    #ds.del_key(get_key_for(task_id, "status"))
+
+    task = models.Task.objects.get(id=task_id)
```
Comment (HeDo88TH) [inline coreplugins/dronedb/api_views.py:277]: This is not working. I need to get the project id from the task id. I'm using the ORM but it seems that no field `id` is found in the `project` object. What am I doing wrong? @pierotofy

**Comment by HeDo88TH** [comment]
I'm now adding the logic to handle edge cases and cleanup. Both the download and the share workflow work! 🚀

**Comment by HeDo88TH** [comment]
This is ready! @pierotofy

**Comment by pierotofy** [comment]
* `celerybeat-schedule.db` should not be included in the repo.

**Comment by pierotofy** [comment]
Using: `https://hub.dronedb.app/r/pierotofy/brighton-beach` as a URL improperly shows 

image

(20 instead of 18)

**Comment by pierotofy** [comment]
I can choose a folder with no input images:

image

**Comment by pierotofy** [comment]
Pressing "DroneDB" flashes a "Fill in your hub credentials" warning, even though my credentials are set (set a network throttle to see it better).

**Comment by pierotofy** [comment]
Can we make the actual DroneDB icon instead of using the font-awesome cloud icon?

image

See the openaerialmap plugin for an example.

**Comment by pierotofy** [comment]
`cameras.json` probably does not need to be uploaded automatically.

But other things like the PDF report or textured model are missing.

Should the user choose which items to upload?

**Comment by pierotofy** [comment]
Really like the ability to browse organizations/datasets... it's a nice touch.

**Comment by pierotofy** [comment]
Overall a few things need to be adjusted, but good progress!

**Comment by pierotofy** [comment]
I'm not sure about the "Go To Dataset" button. I found it a bit confusing, perhaps we could remove it.

image

**Comment by HeDo88TH** [comment]
Ready!

**Comment by markFieldman** [comment]
Thanks for your efforts, @HeDo88TH!
So cool to see such wanted integration 😃

**Comment by pierotofy** [comment]
[quote]
Thanks for this integration! Is it already usable?
[/quote]

Not yet, soon!

**Review by pierotofy — APPROVED** [review]

**Comment by pierotofy** [comment]
Excellent, good to go!

**Pull request merged**